k2/python/k2/fsa_algo.py

```diff
@@ -200,11 +201,7 @@ def compose(a_fsa: Fsa,
     '''
     assert a_fsa.is_cpu()
     assert b_fsa.is_cpu()
-    if not hasattr(a_fsa, 'aux_labels'):
-        return intersect(a_fsa, b_fsa, treat_epsilons_specially)
-
-    if not hasattr(b_fsa, 'aux_labels'):
-        return intersect(a_fsa, b_fsa, treat_epsilons_specially)
+    assert hasattr(a_fsa, 'aux_labels')
```

Comment:
From https://github.com/k2-fsa/k2/pull/554#discussion_r549059387

https://github.com/k2-fsa/k2/blob/d5ce23c44daf814eff40f2a1c28ab210f4c2bee3/k2/python/k2/fsa_algo.py#L148-L150

I was assuming that if either FSA does not have aux_labels, `compose`
is equivalent to `intersect`.

The user has to `invert` the input `a_fsas` himself/herself before
calling `intersect`.

Reply:
I don't think there exist problems with the current code in the master branch.

Reply:
It only does something nontrivial if a_fsa.aux_labels is defined, so if not I'd rather the user just called intersect().

Reply:
If a_fsa.aux_labels are defined and b_fsa.aux_labels are not defined, the current code is wrong. `intersect` will treat a_fsa as being the wrong way round.

Reply:
> intersect will treat a_fsa as being the wrong way round.

That is what I wrote before:

> The user has to invert the input a_fsas himself/herself before calling intersect.

Reply:
Ah I see.  But the interface doesn't make sense in that case-- it's inconsistent and not what the user would expect.